content/courses/advanced-ada/chapters/containers.rst

```diff
@@ -65,6 +65,804 @@ Holder container
     Complete section!
 
 
+.. _Adv_Ada_Aggregate_Aspect:
+
+:ada:`'Aggregate` aspect
+------------------------
+
+.. note::
+
+   This feature was introduced in Ada 2022.
+
```

Comment:
I think you should say what problem this construct solves. One can say that it is for user-defined aggregates, but a natural question for the reader to be wondering is "what do they add that is missing in earlier versions of the language?" so that they can imagine using them in their code. [IIRC they were needed for the sake of defining aggregates for language-defined containers, and then generalized for supporting aggregates for arbitrary user-defined "container" types too.]

Reply:
I'm not sure whether we should really explain that. For all other features of the language —either old or new features—, we're not justifying why they exist. If we do it here, we probably should do it elsewhere as well. This could end up just being too much for this course.

However, if you think a feature or use case associated with the `Aggregate` aspect is missing in this section, I'd be happy to add it here.

(Just to clarify: the note about a feature being introduced in Ada 2022 is just meant for readers that are familiar with Ada 2012 and are now learning about the new features in Ada 2022.)